oauthenticator/generic.py

```diff
@@ -99,52 +95,71 @@ def user_info_to_username(self, user_info):
 
         return username
 
-    async def user_is_authorized(self, auth_model):
-        user_info = auth_model["auth_state"][self.user_auth_state_key]
-        if self.allowed_groups:
-            self.log.info(
-                f"Validating if user claim groups match any of {self.allowed_groups}"
+    def get_user_groups(self, user_info):
+        """
+        Returns a set of groups the user belongs to based on claim_groups_key
+        and provided user_info.
+
+        - If claim_groups_key is a callable, it is meant to return the groups
+          directly.
+        - If claim_groups_key is a nested dictionary key like
+          "permissions.groups", this function returns
+          user_info["permissions"]["groups"].
```

Comment:
Resolves https://github.com/jupyterhub/oauthenticator/pull/594/files#r1209719292 about using / not using a `_` prefix by not adopting a prefix `_`, but helping readers understand that this is not a method part of the common methods referenced by the base class' `authenticate` method as its introduced in GenericOAuthenticator.

```suggestion
          user_info["permissions"]["groups"].

        Note that this method is introduced by GenericOAuthenticator and not
        present in the base class.
```